src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -38,145 +39,150 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Strings.isNullOrEmpty;
 import static java.util.Objects.requireNonNull;
 
 public class SlackEventNotification implements EventNotification {
 
-	public interface Factory extends EventNotification.Factory {
-		@Override
+    private static final Logger LOG = LoggerFactory.getLogger(SlackEventNotification.class);
+    private final EventNotificationService notificationCallbackService;
+    private final Engine templateEngine;
+    private final NotificationService notificationService;
+    private final ObjectMapper objectMapper;
+    private final NodeId nodeId;
+    private final SlackClient slackClient;
+
+    @Inject
+    public SlackEventNotification(EventNotificationService notificationCallbackService,
+                                  ObjectMapper objectMapper,
+                                  Engine templateEngine,
+                                  NotificationService notificationService,
+                                  NodeId nodeId, SlackClient slackClient) {
+        this.notificationCallbackService = notificationCallbackService;
+        this.objectMapper = requireNonNull(objectMapper);
+        this.templateEngine = requireNonNull(templateEngine);
+        this.notificationService = requireNonNull(notificationService);
+        this.nodeId = requireNonNull(nodeId);
+        this.slackClient = requireNonNull(slackClient);
+    }
+
+    /**
+     * @param ctx
+     * @throws EventNotificationException is thrown when send fails
+     */
+    @Override
+    public void execute(EventNotificationContext ctx) throws EventNotificationException {
+        final SlackEventNotificationConfig config = (SlackEventNotificationConfig) ctx.notificationConfig();
+        LOG.debug("SlackEventNotification backlog size in method execute is [{}]", config.backlogSize());
+
+        try {
+            SlackMessage slackMessage = createSlackMessage(ctx, config);
+            slackClient.send(slackMessage, config.webhookUrl());
+        } catch (TemporaryEventNotificationException exp) {
+            //scheduler needs to retry a TemporaryEventNotificationException
+            LOG.error("SlackEventNotification send error for id {} : {}", ctx.notificationId(), exp.toString());
+
+            throw new TemporaryEventNotificationException(String.format("Error sending the SlackNotification message :: %s", exp.getMessage()), exp);
```

Comment:
Since we already get a `TemporaryEventNotificationException`, we can just re-throw it without wrapping it into another `TemporaryEventNotificationException`.